packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/assetModelDataStreamExplorer/assetModelPropertiesExplorer/assetModelPropertiesTable/assetModelPropertiesTable.tsx

```diff
@@ -0,0 +1,119 @@
+import { AssetModelPropertySummary } from '@aws-sdk/client-iotsitewise';
+import { useCollection } from '@cloudscape-design/collection-hooks';
+import Table from '@cloudscape-design/components/table';
+import React, { useEffect } from 'react';
+import { useExplorerPreferences } from '../../../useExplorerPreferences';
+import {
+  ASSET_MODEL_PROPERTIES_TABLE_FILTERING_PROPERTIES,
+  AssetModelPropertiesTablePropertyFilter,
+} from './assetModelPropertiesTableFilter';
+import { AssetModelPropertiesTableEmptyState } from './assetModelPropertiesTableEmptyState';
+import { AssetModelPropertiesTablePagination } from './assetModelPropertiesTablePagination';
+import { AssetModelPropertiesTablePreferences } from './assetModelPropertiesTablePreferences';
+import { AssetModelPropertiesTableColumnDefinitionsFactory } from './assetModelPropertiesTableColumnDefinitionsFactory';
+import { AssetModelPropertiesTableHeader } from './assetModelPropertiesTableHeader';
+import { SelectedAssetModelProperties } from '../../useSelectedAssetModelProperties';
+
+export interface AssetTableProps {
+  onClickNextPage: () => void;
+  onSelectAssetModelProperties: (assetModelProperty: AssetModelPropertySummary[]) => void;
+  assetModelProperties: AssetModelPropertySummary[];
+  selectedAssetModelProperties: SelectedAssetModelProperties;
+  isLoading: boolean;
+  hasNextPage: boolean;
+  isWithoutHeader?: boolean;
+  // client: IoTSiteWiseClient;
+}
+
+export function AssetModelPropertiesTable({
+  assetModelProperties,
+  selectedAssetModelProperties,
+  onClickNextPage,
+  onSelectAssetModelProperties,
+  isLoading,
+  hasNextPage,
+}: // isWithoutHeader,
+// client,
+AssetTableProps) {
+  const [preferences, updatePreferences] = useExplorerPreferences({
+    defaultVisibleContent: ['name'],
+    resourceName: 'asset model properties',
+  });
+
+  const { items, collectionProps, paginationProps, propertyFilterProps, actions } = useCollection(
+    assetModelProperties,
+    {
+      propertyFiltering: {
+        filteringProperties: ASSET_MODEL_PROPERTIES_TABLE_FILTERING_PROPERTIES,
+      },
+      pagination: { pageSize: preferences.pageSize },
+      selection: {
+        defaultSelectedItems: selectedAssetModelProperties,
+        trackBy: ({ id = '' }) => id,
+        keepSelection: true,
+      },
+      sorting: {},
+    }
+  );
+
+  function handleClickNextPage() {
+    onClickNextPage();
+  }
+
+  const selectedPropertiesDependency = JSON.stringify(selectedAssetModelProperties);
+  useEffect(() => {
+    actions.setSelectedItems(selectedAssetModelProperties);
+  }, [selectedPropertiesDependency, actions, selectedAssetModelProperties]);
+
```

Comment:
We cannot use the default selected prop provided by the useCollection hook as the table definition is async since we need to call the iotsitewise api to get the relevant information. This was the only way I could sync the selection once the data is available.